File: matching-learning-api/matching-learning.api/Domain/DTOs/Candidate.cs
using System;
using System.Collections.Generic;
using matching_learning.api.Domain.Enums;

namespace matching_learning.api.Domain.DTOs
{
    public class Candidate : IComparable<Candidate>
    {
        public int Id { get; set; }

        public int DeliveryUnitId { get; set; }

        public DeliveryUnit DeliveryUnit { get; set; }

        public CandidateRelationType RelationType { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Name
        {
            get
            {
                return (this.FirstName + ", " + this.LastName);
            }
        }
        public DocumentType? DocType { get; set; }

        public string DocNumber { get; set; }

        public int? EmployeeNumber { get; set; }

        public bool InBench { get; set; }

        public List<CandidateRoleHistory> RolesHistory { get; set; }

        public int CompareTo(Candidate other)
        {
            return (Name.CompareTo(other.Name));
        }
    }
}

File: matching-learning/src/app/shared/models/candidate.ts
import {DeliveryUnit} from './deliveryUnit';

import {Role} from './role';
import {Evaluation} from './evaluation';
import {Project} from './project';

export interface Candidate {
  id: number;
  deliveryUnitId: number;
  deliveryUnit: DeliveryUnit;
  relationType: number;
  firstName: string;
  lastName: string;
  name: string;
  candidateRoleId: number;
  candidateRole: Role;
  grade: number;
  inBench: boolean;
  docType: number;
  docNumber: string;
  employeeNumber: number;
  project: Project;
  currentProjectId: number;
  CurrentProjectJoin: Date;

  picture: string;
  isActive: boolean;
  gradeDescription: string;
  currentProjectDescription: string;
  currentProjectDuration: string;
  evaluations: Evaluation[];
}
